vhost_user_net/src/lib.rs

```diff
@@ -153,6 +133,53 @@ impl VhostUserNetBackend {
 
         Self::new_with_tap(taps, num_queues, queue_size)
     }
+
+    /// Run vring epoll handler
+    /// Backend could run its specifid vring epoll handler in this way. Such as
+    /// net backend, if it has multiple threads created in backend lib code, all
+    /// the threads will share the same data structure which could cause race
+    /// condition and lead to performance penalty. To create threads by net backend
+    /// itself, and define data structure for each threads to access respectively,
+    /// performance could be improved.
+    pub fn run(&mut self) -> VhostUserBackendResult<()> {
+        let mut tap_set: Vec<(Tap, usize)> = Vec::new();
+        for (i, tap) in self.taps.iter().enumerate() {
+            tap_set.push((tap.clone(), self.num_queues + i));
+        }
+
+        let mut rxs: Vec<RxVirtio> = Vec::new();
+        let mut txs: Vec<TxVirtio> = Vec::new();
+        let mut rx_tap_listenings: Vec<bool> = Vec::new();
+
+        for _ in 0..self.taps.len() {
+            let rx = RxVirtio::new();
+            rxs.push(rx);
+            let tx = TxVirtio::new();
+            txs.push(tx);
+            rx_tap_listenings.push(false);
+        }
+
+        if let Some(vrings) = self.vrings.take() {
+            let mut handler = NetEpollHandler {
+                mem: self.mem.clone(),
+                taps: tap_set,
+                rxs,
+                txs,
+                rx_tap_listenings,
+                epoll_fd: self.epoll_fd,
```

Comment:
Each handler needs its own epoll_fd. You also need a way to map each worker's epoll_fd with one or more vring index, to register the RawFd's when receiving a VHOST_USER_SET_VRING_KICK.

Reply:
Since this PR only handle one worker thread, we could extend later by adding vector<epoll_fd>, and map them with vrings in set_vring_epoll_fd based on backends' requirement.